standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1853,14 +1853,26 @@ private MTable getMTable(String catName, String db, String table) {
       db = normalizeIdentifier(db);
       catName = normalizeIdentifier(catName);
 
-      List<String> lowered_tbl_names = new ArrayList<>(tbl_names.size());
-      for (String t : tbl_names) {
-        lowered_tbl_names.add(normalizeIdentifier(t));
+      List<String> lowered_tbl_names = new ArrayList<>();
+      if(tbl_names != null) {
+        lowered_tbl_names = new ArrayList<>(tbl_names.size());
+        for (String t : tbl_names) {
+          lowered_tbl_names.add(normalizeIdentifier(t));
+        }
       }
 
       query = pm.newQuery(MTable.class);
-      query.setFilter("database.name == db && database.catalogName == cat && tbl_names.contains(tableName)");
-      query.declareParameters("java.lang.String db, java.lang.String cat, java.util.Collection tbl_names");
+
+      if(tbl_names != null && !tbl_names.isEmpty() && tablePattern != null) {
```

Comment:
Can we simplify these conditions to something like this? 

filterString = "database.name == db && database.catalogName == cat";
filterParams = "java.lang.String db, java.lang.String cat";
if(tbl_names != null && !tbl_names.isEmpty()) {
        filterString = filterString + " " + "&& tbl_names.contains(tableName)";
        filterParams = filterParams + " " + "java.util.Collection tbl_names";
      }
if (tablePattern != null) {
        filterString = filterString + " " + "&& tableName.matches(tablePattern)");
        filterParams = filterParams + " " + "java.lang.String tablePattern";
      }